explorer/types.go

```diff
@@ -123,9 +123,32 @@ type SiafundOutput struct {
 	types.SiafundElement
 }
 
-// A FileContract is a types.FileContractElement with added fields for
-// resolved/valid state.
+// A ContractSiacoinOutput is a SiacoinOutput with an added field for its ID.
+type ContractSiacoinOutput struct {
+	ID types.SiacoinOutputID `json:"id"`
+	types.SiacoinOutput
+}
+
+// A FileContract is a storage agreement between a renter and a host. It
+// contains a bidirectional payment channel that resolves as either "valid" or
+// "missed" depending on whether a valid StorageProof is submitted for the
+// contract.
 type FileContract struct {
+	ID                 types.FileContractID    `json:"id"`
+	Filesize           uint64                  `json:"filesize"`
+	FileMerkleRoot     types.Hash256           `json:"fileMerkleRoot"`
+	WindowStart        uint64                  `json:"windowStart"`
+	WindowEnd          uint64                  `json:"windowEnd"`
+	Payout             types.Currency          `json:"payout"`
+	ValidProofOutputs  []ContractSiacoinOutput `json:"validProofOutputs"`
+	MissedProofOutputs []ContractSiacoinOutput `json:"missedProofOutputs"`
+	UnlockHash         types.Hash256           `json:"unlockHash"`
+	RevisionNumber     uint64                  `json:"revisionNumber"`
+}
+
+// A EnhancedFileContract is a FileContractElement with added fields for
+// resolved/valid state, and when the transaction was confirmed and proved.
+type EnhancedFileContract struct {
```

Comment:
Maybe `ExtendedFileContract` is a better name? Otherwise it feels like it should be better ^^